### PR TITLE
En 10588 implement receipts processor

### PR DESCRIPTION
**Pull request by mariusmihaic**
- Implemented receipts processor
- Refactored SCR processor tests

### CODE REVIEW

**Review by AdoAdoAdo — APPROVED** [review]
todo: 
- param order for tests require
- goimports

File: process/receipts/receipts_processor.go [hunk 1]
```diff
@@ -1,18 +1,58 @@
...
 
 // NewReceiptsProcessor creates a new instance of receipts processor
-func NewReceiptsProcessor() (*receiptsProcessor, error) {
-	return &receiptsProcessor{}, nil
+func NewReceiptsProcessor(pubKeyConverter core.PubkeyConverter) (*receiptsProcessor, error) {
+	if check.IfNil(pubKeyConverter) {
+		return nil, covalent.ErrNilPubKeyConverter
+	}
+
+	return &receiptsProcessor{pubKeyConverter: pubKeyConverter}, nil
```
Comment (miiu96) [inline process/receipts/receipts_processor.go:23]: split in multiple lines

File: process/receipts/receipts_processor.go [hunk 2]
```diff
@@ -1,18 +1,58 @@
...
 }
 
 // ProcessReceipts converts receipts data to a specific structure defined by avro schema
-func (rp *receiptsProcessor) ProcessReceipts(transactions map[string]data.TransactionHandler) ([]*schema.Receipt, error) {
-	return nil, nil
+func (rp *receiptsProcessor) ProcessReceipts(receipts map[string]data.TransactionHandler, timeStamp uint64) []*schema.Receipt {
+	allReceipts := make([]*schema.Receipt, 0)
```
Comment (miiu96) [inline process/receipts/receipts_processor.go:28]: allReceipts := make([]*schema.Receipt, 0, len(receipts))

File: process/receipts/receipts_processor.go [hunk 3]
```diff
@@ -1,18 +1,58 @@
...
+		}
+	}
+
+	return allReceipts
+}
+
+func (rp *receiptsProcessor) processReceipt(
+	tx data.TransactionHandler,
+	receiptHash string,
+	timeStamp uint64) *schema.Receipt {
```
Comment (miiu96) [inline process/receipts/receipts_processor.go:43]: add empty line

File: process/receipts/receipts_test.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
+package receipts_test
+
+import (
+	"github.com/ElrondNetwork/covalent-indexer-go"
+	"github.com/ElrondNetwork/covalent-indexer-go/mock"
+	"github.com/ElrondNetwork/covalent-indexer-go/process/receipts"
+	"github.com/ElrondNetwork/covalent-indexer-go/process/utility"
```
Comment (miiu96) [inline process/receipts/receipts_test.go:6]: sort imports

File: process/transactions/sc_processor_test.go [hunk 5]
```diff
@@ -4,12 +4,16 @@ import (
 	"github.com/ElrondNetwork/covalent-indexer-go"
 	"github.com/ElrondNetwork/covalent-indexer-go/mock"
 	"github.com/ElrondNetwork/covalent-indexer-go/process/transactions"
+	"github.com/ElrondNetwork/covalent-indexer-go/process/utility"
```
Comment (miiu96) [inline process/transactions/sc_processor_test.go:6]: sort imports

**Review by miiu96 — CHANGES_REQUESTED** [review]

File: process/receipts/receipts_processor.go (hunk 1 above)
Comment (mariusmihaic) [reply process/receipts/receipts_processor.go:23]: Done

File: process/receipts/receipts_processor.go (hunk 2 above)
Comment (mariusmihaic) [reply process/receipts/receipts_processor.go:28]: Done

File: process/receipts/receipts_processor.go (hunk 3 above)
Comment (mariusmihaic) [reply process/receipts/receipts_processor.go:43]: Done

File: process/receipts/receipts_test.go (hunk 4 above)
Comment (mariusmihaic) [reply process/receipts/receipts_test.go:6]: Done

File: process/transactions/sc_processor_test.go (hunk 5 above)
Comment (mariusmihaic) [reply process/transactions/sc_processor_test.go:6]: Done

**Review by AdoAdoAdo — COMMENTED** [review]
fixes to be done on EN-10606-Review_fixes

**Pull request merged**